### PR TITLE
Add workflow step to publish a release

### PR DESCRIPTION
**Pull request by jubalm**
The release output would be similar to this format

| Screenshot 2024-02-24 at 11 17 48 PM |
| -- |
| *Sample Release* |

### CODE REVIEW

File: .github/workflows/ipfs-deploy.yml [hunk 1]
```diff
@@ -65,3 +65,71 @@ jobs:
       - name: Publish to nft.storage
         run: |
           docker run -e NFTSTORAGE_API_KEY=${{ secrets.NFTSTORAGE_API_KEY }} $IMAGE_RELEASE_ID nft.storage
+
+      - name: Create a release
+        if: startsWith(github.ref, 'refs/tags/')
+        env:
+          REF_NAME: ${{ github.ref_name }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
```
Comment (MicahZoltu) [inline .github/workflows/ipfs-deploy.yml:73]: These are both available to all workflow steps automatically.  No need to set them in an `env` section.  Secrets are special and must be specified in the `env` section to be available in steps.
```suggestion
```
Note: You'll need to rename to `GITHUB_REF_NAME` elsewhere.

File: .github/workflows/ipfs-deploy.yml [hunk 2]
```diff
@@ -65,3 +65,71 @@ jobs:
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Markdown template for the release notes
+          RELEASE_NOTE_TEMPLATE=$(cat << EOF
+          #### IPFS Hash
+
+          \`\`\`
+          $IPFS_HASH
+          \`\`\`
+
```
Comment (MicahZoltu) [inline .github/workflows/ipfs-deploy.yml:83]: No need to put it on a line by itself, all of the links have it.
```suggestion
```

File: .github/workflows/ipfs-deploy.yml [hunk 3]
```diff
@@ -65,3 +65,71 @@ jobs:
...
+          [https://$IPFS_HASH.ipfs.w3s.link](https://$IPFS_HASH.ipfs.w3s.link)
+          EOF
+          )
+
+          # Generate payload for creating a new release
+          PAYLOAD_TEMPLATE=$(cat <<EOF
+          {
+            "name": "$REF_NAME",
+            "tag_name": "$REF_NAME",
+            "target_commitish": "main",
```
Comment (MicahZoltu) [inline .github/workflows/ipfs-deploy.yml:101]: I don't think this is correct?  We are releasing from tags, not from `main`.  I'm not sure what should be here, but it feels like `main` is wrong.

File: .github/workflows/ipfs-deploy.yml [hunk 4]
```diff
@@ -65,3 +65,71 @@ jobs:
...
+          [https://$IPFS_HASH.ipfs.cf-ipfs.com](https://$IPFS_HASH.ipfs.cf-ipfs.com)
+          [https://$IPFS_HASH.ipfs.w3s.link](https://$IPFS_HASH.ipfs.w3s.link)
+          EOF
+          )
+
+          # Generate payload for creating a new release
+          PAYLOAD_TEMPLATE=$(cat <<EOF
+          {
+            "name": "$REF_NAME",
+            "tag_name": "$REF_NAME",
```
Comment (MicahZoltu) [inline .github/workflows/ipfs-deploy.yml:100]:
```suggestion
            "name": "$GITHUB_REF_NAME",
            "tag_name": "$GITHUB_REF_NAME",
```

File: .github/workflows/ipfs-deploy.yml [hunk 5]
```diff
@@ -65,3 +65,71 @@ jobs:
...
+          RESPONSE_BODY=${RESPONSE//$RESPONSE_CODE/}
+
+          # Successful creation will return a status 201 (Created), otherwise show the error
+          if [ "$RESPONSE_CODE" -ne  201 ]; then
+            ERROR_MESSAGE=$(echo "$RESPONSE_BODY" | jq '.message')
+            echo "HTTP $RESPONSE_CODE - $ERROR_MESSAGE"
+            exit 1
+          fi
+
+          echo "Release ($REF_NAME) successfully created"
```
Comment (MicahZoltu) [inline .github/workflows/ipfs-deploy.yml:135]:
```suggestion
          echo "Release ($GITHUB_REF_NAME) successfully created"
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
A few changes suggested, recommend reviewing to make sure I didn't introduce bugs (e.g., by not updating a variable name somewhere).

File: .github/workflows/ipfs-deploy.yml (hunk 2 above)
Comment (KillariDev) [reply .github/workflows/ipfs-deploy.yml:83]: I like that its clearly somewhere and you don't need to dig it on rest of the others

**Comment by KillariDev** [comment]
ipfs hash being nicely copyable is a nice touch!

File: .github/workflows/ipfs-deploy.yml (hunk 2 above)
Comment (MicahZoltu) [reply .github/workflows/ipfs-deploy.yml:83]: Hmm, I guess having it be copyable (as you mentioned below) does give an advantage.  Will retract this suggestion.

File: .github/workflows/ipfs-deploy.yml (hunk 3 above)
Comment (jubalm) [reply .github/workflows/ipfs-deploy.yml:104]: Removed the payload value as it's not needed as the tag is already created.

Screenshot 2024-02-27 at 2 58 03 PM

https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28

File: .github/workflows/ipfs-deploy.yml [hunk 6]
```diff
@@ -65,3 +65,71 @@ jobs:
       - name: Publish to nft.storage
         run: |
           docker run -e NFTSTORAGE_API_KEY=${{ secrets.NFTSTORAGE_API_KEY }} $IMAGE_RELEASE_ID nft.storage
+
+      - name: Create a release
+        if: startsWith(github.ref, 'refs/tags/')
```
[the remark this answers is not in the document]
Comment (jubalm) [reply .github/workflows/ipfs-deploy.yml]: applied on https://github.com/DarkFlorist/lunaria/pull/261/commits/72fe8eac662858e8ab713b42b4c7907c7263273f

File: .github/workflows/ipfs-deploy.yml (hunk 1 above)
Comment (jubalm) [reply .github/workflows/ipfs-deploy.yml]: applied on https://github.com/DarkFlorist/lunaria/pull/261/commits/72fe8eac662858e8ab713b42b4c7907c7263273f

File: .github/workflows/ipfs-deploy.yml (hunk 4 above)
Comment (jubalm) [reply .github/workflows/ipfs-deploy.yml]: applied on https://github.com/DarkFlorist/lunaria/pull/261/commits/72fe8eac662858e8ab713b42b4c7907c7263273f

File: .github/workflows/ipfs-deploy.yml (hunk 5 above)
Comment (jubalm) [reply .github/workflows/ipfs-deploy.yml]: applied on https://github.com/DarkFlorist/lunaria/pull/261/commits/72fe8eac662858e8ab713b42b4c7907c7263273f

**Review by MicahZoltu — APPROVED** [review]

**Pull request merged**